main.go

```diff
@@ -12,6 +24,125 @@ func init() {
 	_ = flag.Set("logtostderr", "true")
 }
 
+func GetCurrentNode(clientset kubernetes.Clientset) (*corev1.Node, error) {
+	nodeName := os.Getenv("NODE_NAME")
+	if nodeName == "" {
+		return nil, errors.New("Environment variable NODE_NAME is not set")
+	}
+
+	node, err := clientset.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get node: %w", err)
+	}
+
+	return node, nil
+}
+
+func UpdateNodeLabels(
+	clientset *kubernetes.Clientset,
+	instanceData *metadata.InstanceData,
+) error {
+	if instanceData == nil {
+		return fmt.Errorf("instance data received from Linode metadata service is nil")
+	}
+
+	node, err := GetCurrentNode(*clientset)
+	if err != nil {
+		return fmt.Errorf("failed to get the node: %w", err)
+	}
+
+	klog.Infof("Updating node labels with Linode instance data: %v", instanceData)
+
+	node.Labels["linode_label"] = instanceData.Label
+	node.Labels["linode_id"] = strconv.Itoa(instanceData.ID)
+	node.Labels["linode_region"] = instanceData.Region
+	node.Labels["linode_type"] = instanceData.Type
+	node.Labels["linode_host"] = instanceData.HostUUID
+
+	_, err = clientset.CoreV1().Nodes().Update(context.TODO(), node, metav1.UpdateOptions{})
```

Comment:
i know the watcher wont emit an event unless the data changed but you should also gate the Update with a diff check as well, no point spamming the Updates if there are no changes to any of the fields. Just if check instanceData to the node.Labels and return nil if nothing has changed should be enough

Reply:
It has been implemented, thanks!